pkg/openstack/routes.go

```diff
@@ -59,12 +67,93 @@ func NewRoutes(os *OpenStack, network *gophercloud.ServiceClient, atomicRoutes b
 		return nil, errors.ErrNoRouterID
 	}
 
-	return &Routes{
+	routes := Routes{
 		network:             network,
 		os:                  os,
 		atomicRoutes:        atomicRoutes,
 		allowedAddressPairs: allowedAddressPairs,
-	}, nil
+	}
+
+	err := routes.getOrCreateNodeSecurityGroup(os.routeOpts.RouterID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &routes, nil
```

Comment:
I strongly think this shouldn't be done in the initialization of the controller, but rather on `CreateRoute()`, each time checking if SG exists so that it's existence can be reconciled if deleted. We can cache it's ID if needed.